Dockerfile

```diff
@@ -15,29 +12,55 @@ RUN cd /root/bcc && \
     /usr/lib/pbuilder/pbuilder-satisfydepends && \
     PARALLEL=$(nproc) ./scripts/build-deb.sh release
 
-FROM ubuntu:bionic
+RUN cd /root && git clone https://github.com/libbpf/libbpf.git && \
+    cd /root/libbpf && git checkout 030ff87857090ae5c9d74859042d05bfb3b613a2  && cd src && mkdir build root && BUILD_STATIC_ONLY=y OBJDIR=build DESTDIR=root make install
+
+FROM ubuntu:20.04 as compiler
+ENV DEBIAN_FRONTEND=noninteractive
 
 RUN apt-get update && \
-    apt-get install -y --no-install-recommends git build-essential libelf1 software-properties-common
+    apt-get install -y --no-install-recommends build-essential pkg-config libelf-dev libz-dev software-properties-common clang
 
 RUN add-apt-repository ppa:longsleep/golang-backports && \
     apt-get install -y --no-install-recommends golang-1.17-go
 
 ENV PATH="/usr/lib/go-1.17/bin:$PATH"
 
 COPY --from=builder /root/bcc/libbcc_*.deb /tmp/libbcc.deb
+COPY --from=builder /root/bcc/libbpf-tools  /tmp/libbpf-tools
+COPY --from=builder /root/libbpf/ /tmp/libbpf
 
+RUN cd /tmp/libbpf/src && make install
 RUN dpkg -i /tmp/libbcc.deb
 
 COPY ./ /go/ebpf_exporter
 
-RUN cd /go/ebpf_exporter && \
-    GOPROXY="off" GOFLAGS="-mod=vendor" go install -v -ldflags=" \
+RUN cd /go/ebpf_exporter && CC=gcc CGO_CFLAGS="-I /usr/include/bpf" CGO_LDFLAGS="-lbpf" GOPATH="" GOPROXY="off" GOFLAGS="-mod=vendor" go install -v -ldflags=" \
     -X github.com/prometheus/common/version.Version=$(git describe) \
     -X github.com/prometheus/common/version.Branch=$(git rev-parse --abbrev-ref HEAD) \
     -X github.com/prometheus/common/version.Revision=$(git rev-parse --short HEAD) \
     -X github.com/prometheus/common/version.BuildUser=docker@$(hostname) \
     -X github.com/prometheus/common/version.BuildDate=$(date --iso-8601=seconds) \
     " ./cmd/ebpf_exporter
+RUN cd /go/ebpf_exporter/examples/CORE && make all
+
+
+FROM ubuntu:20.04
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends build-essential pkg-config libelf-dev libz-dev software-properties-common
+
+COPY --from=builder /root/bcc/libbcc_*.deb /tmp/libbcc.deb
+COPY --from=builder /root/bcc/libbpf-tools  /tmp/libbpf-tools
+COPY --from=builder /root/libbpf/ /tmp/libbpf
+COPY --from=compiler /root/go/bin/ebpf_exporter /ebpf_exporter
+COPY --from=compiler /go/ebpf_exporter/examples /examples
+
+RUN cd /tmp/libbpf/src && make install
+RUN dpkg -i /tmp/libbcc.deb
+RUN find /tmp -name "*.git*"|xargs rm -Rf
```

Comment:
Why remove stuff here?

Reply:
`git` info is treated as vulnerability in some companies's image security policy, so remove this.

Reply:
For people that do this, there's always an option to build from source. We don't need to do any cleanup in this image, as it's not intended as a distributable artifact. There are only two things of interest here:

* The image should build a binary
* The binary should be compatible with older libc (as mentioned in the very first comment)